src/plugins/dataflow/nodes/dataflow-node-plot.tsx

```diff
@@ -59,15 +97,18 @@ function lineData(node: any) {
       recentValues.forEach((val: any) => {
         if (isFinite(val)) {
           chData.push(val);
-          dsMax = Math.max(dsMax, val);
-          dsMin = Math.min(dsMin, val);
         }
       });
+      const localMax = Math.max(...chData);
+      node.data.dsMax = ((node.data.dsMax !== undefined) ? Math.max(localMax, node.data.dsMax) : localMax);
+      const localMin = Math.min(...chData);
+      node.data.dsMin = ((node.data.dsMin !== undefined) ? Math.min(localMin, node.data.dsMin) : localMin);
       dataset.data = chData;
       chartDataSets.push(dataset);
     }
   });
-  stepY = (dsMax - dsMin) / 2;
+
+  stepY = (node.data.dsMax  - node.data.dsMin) / 2;
```

Comment:
```suggestion
  stepY = (maxY(node) - minY(node)) / 2;
```
I posted this comment earlier, but I'm guessing it was hidden because it was part of a resolved comment. But you should change this code unless there's a good reason not to that I'm not seeing.

It's worth mentioning that this isn't just more concise, it's functionally different. With the existing code, if the user pushes the + or - button, the graph will have node.data.tickMax/tickMin as its max and min, but stepY will still be defined by node.data.dsMax/dsMin. The change I'm suggesting will make them both defined by the same variables, either tickMax/tickMin or dsMax/dsMin. Given how small a change this is, I think it's definitely worth doing it right.

Reply:
Ah right yeah I understand how this is functionally different, I just didn't see it in the earlier thread. committed.